lib/operations/add_user.js

```diff
@@ -1,186 +1,113 @@
 'use strict';
 
 const Aspect = require('./operation').Aspect;
-const defineAspects = require('./operation').defineAspects;
-const OperationBase = require('./operation').OperationBase;
 const CommandOperation = require('./command');
-const applyWriteConcern = require('../utils').applyWriteConcern;
-const CONSTANTS = require('../constants');
-const count = require('./collection_ops').count;
+const defineAspects = require('./operation').defineAspects;
 const crypto = require('crypto');
 const handleCallback = require('../utils').handleCallback;
-const loadDb = require('../dynamic_loaders').loadDb;
-const MongoError = require('../core').MongoError;
-const ReadPreference = require('../core').ReadPreference;
 const toError = require('../utils').toError;
-const updateOne = require('./collection_ops').updateOne;
+const WriteConcern = require('../write_concern');
 
-class AddUserOperation extends OperationBase {
+class AddUserOperation extends CommandOperation {
   constructor(db, username, password, options) {
-    super(options);
+    super(db, {}, options);
 
-    this.db = db;
     this.username = username;
     this.password = password;
+    this.finalOptions = Object.assign({}, options);
```

Comment:
This is just a copy, and you're already storing the options by passing it to the base class. Instead of saving off a copy as a property of the operation, consider instead just defining this as a local instance variable in `_buildCommand`